libkmod/libkmod-index.c

```diff
@@ -628,18 +628,14 @@ struct index_mm_value {
 	const char *value;
 };
 
-struct index_mm_value_array {
-	struct index_mm_value *values;
-	size_t len;
-};
-
 struct index_mm_node {
 	struct index_mm *idx;
 	const char *prefix; /* mmap'ed value */
-	struct index_mm_value_array values;
 	unsigned char first;
 	unsigned char last;
-	uint32_t children[];
+	const void *children; /* mmap'ed value */
```

Comment:
No, this makes a trap you've fallen into as well: The pointers are not aligned for anything, so it's definitely not just an access to a `uint32_t` here. Also, these numbers are in big endian format, not host. It's better to make it clear that nobody should ever think about accessing these pointers like that.

Reply:
What is stopping us from swapping the endinaness on demand? Would it be slower?

Reply:
That's what we do with this PR. :)

Reply:
I'm confused, are we talking about the `void` to `uint32_t` suggestion or something else where... It feels like the initial reply is quoting the wrong suggestion.

Reply:
I mean, what we do now is:
- Pass the `void` pointer (not turning it into `uint32_t` to show that it shouldn't be accessed directly to get `uint32_t`s because they are unaligned big endian numbers) to `read_u32_mm`
- `read_u32_mm` is called on demand if the child offset is actually needed

Reply:
Right, so the type stays as-is. Fair enough.